perun-core/src/main/java/cz/metacentrum/perun/core/entry/MembersManagerEntry.java

```diff
@@ -1459,6 +1460,22 @@ public void sponsorMembers(PerunSession session, List<Member> sponsored, User sp
 		}
 	}
 
+	@Override
+	public void copySponsoredMembers(PerunSession session, List<MemberWithSponsors> sponsored, User copyFrom, User copyTo, boolean copyValidity, LocalDate validityTo) throws SponsorshipDoesNotExistException, UserNotInRoleException, AlreadySponsorException, PrivilegeException, MemberNotSponsoredException {
+		for (MemberWithSponsors member : sponsored) {
```

Comment:
if i'm not mistaken, it seems MemberWithSponsors object is not retrieved from backend, but relies on the object provided in the request - we should then check if that information is valid, same for `validityTo`. It would be also more user-friendly for API callers to specify just the member id instead of whole MemberWithSponsor object.